src/OrchardCore.Modules/OrchardCore.OpenId/Configuration/OpenIdValidationConfiguration.cs

```diff
@@ -138,49 +118,19 @@ public void Configure(string name, JwtBearerOptions options)
                     return;
                 }
 
+                // If the authority was specified in the server options, use it as the issuer name.
+                options.Issuer = configuration.Authority;
+
                 // When the server is another tenant, don't allow the current tenant
                 // to choose the valid audiences, as this would otherwise allow it
                 // to validate/introspect tokens meant to be used with another tenant.
-                options.TokenValidationParameters.ValidAudience = OpenIdConstants.Prefixes.Tenant + _shellSettings.Name;
+                options.Audiences.Add(OpenIdConstants.Prefixes.Tenant + _shellSettings.Name);
                 options.TokenValidationParameters.IssuerSigningKeys = await service.GetSigningKeysAsync();
-
-                // If an authority was explicitly set in the OpenID server options,
-                // prefer it to the dynamic tenant comparison as it's more efficient.
-                if (configuration.Authority != null)
-                {
-                    options.TokenValidationParameters.ValidIssuer = configuration.Authority.AbsoluteUri;
-                }
-                else
-                {
-                    options.TokenValidationParameters.IssuerValidator = (issuer, token, parameters) =>
```

Comment:
None of this gnarly stuff will be required in 3.0, as the OpenIddict validation handler will support JWT and infer the issuer from the request host.